File: Codeforces/Global-Rounds/14/Phoenix-and-Puzzle/main.cpp
// https://codeforces.com/contest/1515/problem/B

#include <bits/stdc++.h>

using namespace std;

bool isSquare(int n) {
    int y = sqrt(n);
    return y*y == n;
}

string itsPossible(int n) {
    if (!(n%2) && isSquare(n/2))
        return "YES";
    
    if (!(n%4) && isSquare(n/4))
        return "YES";
    
    return "NO";
}

int main () {
    cin.tie(0);
    ios_base::sync_with_stdio(0);

    int t; 
    cin >> t;

    while(t--){
        int n;
        cin >> n; 
        cout << itsPossible(n) << endl;
    }

    return 0;
}

File: Codeforces/Global-Rounds/14/Phoenix-and-Puzzle/main.py
# https://codeforces.com/contest/1515/problem/B

import math

def is_square (n):
    y = int(math.sqrt(n))
    return y*y == n

def its_possible (n):
    if not n%2 and is_square(n // 2):
        return "YES"
    if not n%4 and is_square(n // 4):
        return "YES"

    return "NO"

def main():
    for i in range(int(input())):
        print(its_possible(int(input())))

main()
